server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -2866,7 +2892,11 @@ public UserVm rebootVirtualMachine(RebootVMCmd cmd) throws InsufficientCapacityE
             throw new InvalidParameterValueException("Unable to find service offering: " + serviceOfferingId + " corresponding to the vm");
         }
 
-        UserVm userVm = rebootVirtualMachine(CallContext.current().getCallingUserId(), vmId);
+        Boolean enterSetup = cmd.getBootIntoSetup();
+        if (! (enterSetup == null || HypervisorType.VMware.equals(vmInstance.getHypervisorType()))) {
```

Comment:
For readability can you change it to: enterSetup != null && hypervisor != Vmware?

Reply:
would `enterSetup != null && !HypervisorType.VMware.equals(vmInstance.getHypervisorType())` be acceptable as well?

Reply:
Sure